lib/src/components/NxThreatCounter/__tests__/NxThreatCounter.test.tsx

```diff
@@ -7,47 +7,99 @@
 import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
 
 import NxThreatCounter, { Props } from '../NxThreatCounter';
+import {capitalize} from '@material-ui/core';
 
 describe('NxThreatCounter', function() {
   const minimalProps = {
         criticalCount: 66,
         severeCount: 55,
         moderateCount: 44,
-        lowCount: 33
+        lowCount: 33,
+        noneCount: 22
       },
+      names = ['critical', 'severe', 'moderate', 'low', 'none'],
       getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxThreatCounter, minimalProps);
 
+  const getCounts = (value: number | null | undefined) => {
+    const props: {[index: string]:number | null | undefined} = {
+      criticalCount: value,
+      severeCount: value,
+      moderateCount: value,
+      lowCount: value,
+      noneCount: value
+    };
+    return props;
+  };
+
+  it('renders nothing if all five indicators are undefined', function() {
+    const component = getShallowComponent(getCounts(undefined));
+    expect(component).toBeEmptyRender();
+  });
+
+  it('renders nothing if all five indicators are null', function() {
+    const component = getShallowComponent(getCounts(null));
+    expect(component).toBeEmptyRender();
+  });
+
   it('renders the container with the expected default classes', function() {
     expect(getShallowComponent().find('.nx-threat-counter-container')).toExist();
   });
 
-  it('renders all four indicators', function() {
+  const checkEachCountRendersAlone = (value: null | undefined) => {
+    names.forEach(name => {
+      const props = getCounts(value);
+      const countName = name + 'Count';
+      props[countName] = 0;
+      const container = getShallowComponent(props).find('.nx-threat-counter-container');
+      expect(container).toExist();
+      const counts = container.find('div');
+      expect(counts.length).toBe(1);
+      const count = counts.at(0);
+      expect(count.find('.nx-threat-counter__count')).toHaveText('0');
+      expect(count.find('.nx-threat-counter__text')).toHaveText(capitalize(name));
```

Comment:
This shouldn't be working, the text isn't capitalized at the DOM level but at the styling level

Reply:
> This shouldn't be working, the text isn't capitalized at the DOM level but at the styling level

Oh nvm I was thinking fully uppercased not capitalized.  At any rate, this is a bit too moch logic for a test IMO.  I'd rather it all be spelled out in a more hardcoded way

Reply:
> Oh nvm I was thinking fully uppercased not capitalized. At any rate, this is a bit too much logic for a test IMO. I'd rather it all be spelled out in a more hardcoded way

Attempted to make it slightly more explicit https://github.com/sonatype/sonatype-react-shared-components/pull/418/commits/bbbc191c18b7ba8b720c3b782cff9e474889c15f or I can break it into separate tests if you prefer `it('renders only critical if the others are undefined') ...`

Reply:
> Attempted to make it slightly more explicit bbbc191 or I can break it into separate tests if you prefer it('renders only critical if the others are undefined') ...

I'd have a test like `it('renders an .nx-threat-counter--critical when criticalCount is specified'` and then similarly for the other four levels.  And then one with all of them together.

Reply:
> I'd have a test like `it('renders an .nx-threat-counter--critical when criticalCount is specified'` and then similarly for the other four levels. And then one with all of them together.

Done in https://github.com/sonatype/sonatype-react-shared-components/pull/418/commits/c296dfb05ada74f6812ddba6a75396b9945f6d76